lib/ansible/module_utils/net_tools/nios/api.py

```diff
@@ -357,7 +355,7 @@ def get_object_ref(self, module, ib_obj_type, obj_filter, ib_spec):
                 else:
                     test_obj_filter = dict([('name', name), ('view', obj_filter['view'])])
             else:
-                test_obj_filter = dict([('name', name)])
+                test_obj_filter = obj_filter
```

Comment:
Can you please revert following changes as the fix for the respective change is merged via PR #48859. You can keep the other changes and I'll fix the nios_container additional parameter issue so that the shippable won't fail.

Reply:
@justjais, I have pulled out all changes I made to the api.py file. The only thing left in this PR is just the added integration tests.

Reply:
@coreywan We are in process of setting complete `nios` instance for unit testing purpose and that will help to remove the dependency over `nios_container`. So, please wait if it'll take more than expected time I'll make the changes to `nios_container` so that you can be able to merge this PR. Thanks